python/cisa_risk_meter/build_cisa_risk_meter.py

```diff
@@ -145,11 +132,49 @@ def search_vulns_for_cve_id(base_url, headers, cve_id):
 
     # Check for all other errors.
     if response.status_code != 200:
-        process_http_error(f"Search Vulnerabilities API with query string {q_str} ", response, search_url)
+        process_http_error(f"Search Vulnerabilities API", response, search_url)
         sys.exit(1)
     
     resp_json = response.json()
-    return resp_json['vulnerabilities']
+    return(resp_json['vulnerabilities'], resp_json['meta'])
+
+# CVE ID vulnerability search.
+def search_vulns_for_cve_id(base_url, headers, cve_id):
+    logging.info(f"Searching vulnerabilities for CVE ID {cve_id}")
+
+    # Make a searchable CVE ID.
+    cve_str = cve_id[0:4].upper()
+    if cve_str != "CVE-":
+        print(f"Input is not a CVE, {cve_str}")
+        logging.error(f"Internal error: Expecting CVE, but it was {cve_str}")
+        sys.exit(1)
+
+    cve_id_str = cve_id[4:]
+    
+    # Assemble the Search Vulnerability URL.
+    q_str = f"q=cve:{cve_id_str}"
+    search_url = f"{base_url}vulnerabilities/search?{q_str}&per_page={SEARCH_PAGE_SIZE}"
+
+    # Check for the one page case.
+    (vulnerabilities, meta_info) = search_vulns(search_url, headers)
+    if meta_info['page'] == 1:
+        return vulnerabilities
+
+    # Multi-page case.  First append vulnerabilitie from page one.
+    vulns = []
+    vulns.append(vulnerabilities)
+    max_pages = meta_info['pages']
+
+    # Loop through available pages colllecting vulnerabilities.
+    page_num = 2
+    while page_num <= max_pages:
+        search_url += f"&page={page_num}"
```

Comment:
Just to confirm that if the `max_pages` is 3, would the `search_url` contain `&page=1&page=2&page=3` and is it what we wanted?

Reply:
No.  Thanks for catching this concatenation error.

Reply:
Made changes.